gufe/tests/test_tokenization.py

```diff
@@ -211,6 +215,30 @@ def test_notequal_different_type(self):
 
         assert l1 != l2
 
+    @pytest.mark.parametrize('level', ["DEBUG", "INFO", "CRITICAL"])
+    def test_logging(self, level):
+        stream = io.StringIO()
+        handler = logging.StreamHandler(stream)
+        fmt = logging.Formatter("%(name)s - %(levelname)s - %(message)s")
```

Comment:
Can we go with pipes as the delimiter here? Less commonly found in messages, so makes raw log parsing simpler downstream.

Also, can we put the `levelname` first? Might be able to set padding as well so the result looks something like:

```
DEBUG    | gufe.transformations.Transformation | eb839ae80d905adee778c32f516944ad | I'm a silly goose.
INFO     | gufe.chemicalsystem.ChemicalSystem | e1cb9ce41e88ee474cf5b962c9388159 | No, I am in fact, the silly goose.
CRITICAL | gufe.network.AlchemicalSystem       | 54aa4f41f09cb07ebe6febdc65f1c278 | I AM THE SILLIEST GOOSE.

```

Here we have `levelname | name | gufe-key token | message`.

We may also want a timestamp out front, but this may only be desirable for non-STDOUT handlers.

Reply:
This is a format that's only used in this specific test. End user can set any desired format in logging configuration.

Reply:
Ah good point! Agree that this is best handled at the application layer.

Reply:
@dwhswenson what would be the format string to pull the gufe key out of an object?  (i.e. what formatter do I set to get Dotson's example above?)

Reply:
Hmm, I think that's why I thought of having the gufekey be in the name of the logger (one per instance), but then equally it becomes impossible(?) to address the logger via `getLogger()` as it has a random name.

Reply:
No, you could have the logger name include the gufe key. That's not a problem. But then you have a lot more loggers in the logger registry. This could make logging slower; I'm not sure how the backend for that works, but I know the typical usage is 1 logger per module or, at most, 1 per class, so it wouldn't surprise me if 1 per instance is a Really Bad Idea™.

Reply:
We can now inject the gufe key. Turns out I didn't need to write my own class to handle the injection, because there's `logging.LoggerAdapter` for exactly this purpose.